algorithms.go

```diff
@@ -68,7 +70,7 @@ var hashToDef = map[crypto.Hash]struct {
 	// Note that the forbidden hashes have an invalid 'new' function.
 	crypto.MD4:         {md4String, func(key []byte) (hash.Hash, error) { return nil, nil }},
 	crypto.MD5:         {md5String, func(key []byte) (hash.Hash, error) { return nil, nil }},
-	crypto.SHA1:        {sha1String, func(key []byte) (hash.Hash, error) { return nil, nil }},
+	crypto.SHA1:        {sha1String, func(key []byte) (hash.Hash, error) { return sha1.New(), nil }},
```

Comment:
nit: do you mind adding a comment for this line that indicates the only reason this is enabled is because of the linked golang issue you shared in the PR description? It would be much appreciated as a reminder to our future selves.

Reply:
Sure, resolved now :)